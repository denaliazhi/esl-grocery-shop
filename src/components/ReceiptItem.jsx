/**
 * This component renders details for an item in the receipt.
 */
import { capitalize } from "../utils.js";

export default function ReceiptItem({ item, handleMouse }) {
  return (
    <div onMouseOver={handleMouse}>
      <p>{item.count}</p>
      <div>
        <p>{capitalize(item.name)},</p>
        <p>
          {item.unitPrice.toFixed(2)} / {item.displayUnit}
        </p>
      </div>
      <p>{item.totalCost.toFixed(2)}</p>
    </div>
  );
}
