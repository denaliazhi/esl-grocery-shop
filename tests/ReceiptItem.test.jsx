import { vi, describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";

import ReceiptItem from "../src/components/ReceiptItem";
import userEvent from "@testing-library/user-event";

describe("Receipt item", () => {
  test("renders expected content", () => {
    const item = {
      id: "x123",
      name: "apple",
      unitPrice: 1.23,
      displayUnit: "each",
      section: "produce",
      count: 1,
      totalCost: 1.23,
    };
    const { container } = render(
      <ReceiptItem item={item} handleMouse={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  test("on hover, calls handleMouse", async () => {
    const user = userEvent.setup();
    const handleMouse = vi.fn();
    const item = {
      id: "x123",
      name: "apple",
      unitPrice: 1.23,
      displayUnit: "each",
      section: "produce",
      count: 1,
      totalCost: 1.23,
    };
    render(<ReceiptItem item={item} handleMouse={handleMouse} />);
    const renderedItem = screen.getByRole("listitem");
    await user.hover(renderedItem);

    expect(handleMouse).toHaveBeenCalledOnce();
  });
});
