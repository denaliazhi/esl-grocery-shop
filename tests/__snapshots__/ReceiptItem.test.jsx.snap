// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Receipt item > renders expected content 1`] = `
<div>
  <li>
    <p>
      1
    </p>
    <div>
      <p>
        Apple
        ,
      </p>
      <p>
        1.23
         / 
        each
      </p>
    </div>
    <p>
      1.23
    </p>
  </li>
</div>
`;
